CMakeLists.txt

```diff
@@ -7,656 +7,75 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 include(ExternalProject)
 include(GNUInstallDirs)
 
-# Define Options
-option(BUILD_PYTHON "Include Python" ON)
-option(BUILD_QT "Include Qt" ON)
-option(BUILD_SIP "Include SIP" ON)
-option(BUILD_PYQT "Include PyQt" ON)
-option(BUILD_NUMPY "Include Numpy" ON)
-option(BUILD_SCIPY "Include Scipy" ON)
-option(BUILD_PYSERIAL "Include PySerial" ON)
-option(BUILD_NUMPY_STL "Include Numpy-STL" ON)
-option(BUILD_ZEROCONF "Include python-zeroconf" ON)
+# find_package(CuraBuildEnvironment)
+include(BuildPlatformDetection)
+include(GetFromEnvironmentOrCache)
+include(SetProjectDependencies)
 
-if(WIN32)
-    option(BUILD_64BIT "Create a 64-bit build" OFF)
-endif()
-
-# Create variables to simplify OS management
-set(BUILD_OS_OSX OFF)
-set(BUILD_OS_LINUX OFF)
-set(BUILD_OS_WINDOWS OFF)
-set(BUILD_OS_WIN32 OFF)
-set(BUILD_OS_WIN64 OFF)
-
-if(CYGWIN)
-    message(FATAL_ERROR "Cygwin is not supported")
-endif()
-
-if(APPLE)
-    set(BUILD_OS_OSX ON)
-elseif(WIN32)
-    set(BUILD_OS_WINDOWS ON)
-    if(BUILD_64BIT)
-        set(BUILD_OS_WIN64 ON)
-    else()
-        set(BUILD_OS_WIN32 ON)
-    endif()
-elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(BUILD_OS_LINUX ON)
-else()
-    message(FATAL_ERROR "Unsupported Operating System: ${CMAKE_SYSTEM_NAME}")
-endif()
-
-# Explicitly disable unsupported options
-if(BUILD_OS_WINDOWS)
-    set(BUILD_PYTHON OFF)
+option(BUILD_PACKAGE "Create a package with the build of Cura" ON)
+option(SIGN_PACKAGE "Perform signing of the created package. Implies BUILD_PACKAGE" ON)
```

Comment:
Could be worth to look into cmake_dependent_option: https://cmake.org/cmake/help/v3.0/module/CMakeDependentOption.html